packages/web-components/fast-element/src/components/fast-definitions.ts

```diff
@@ -94,10 +94,23 @@ export class FASTElementDefinition<
         return this.platformDefined;
     }
 
+    /**
+     * The prefix of the custom element's name
+     */
+    public prefix: string;
```

Comment:
With interpolation readily available for use within `name`, this seems like a heavy-handed approach to me. Is there a reason something like what's represented here wouldn't just work without needing add additional properties to the definition itself? https://stackblitz.com/edit/typescript-npcwww?file=button%2Fbutton.ts

Reply:
In the case of a library built with FE if the library is exporting definitions by using `compose`, the name is already defined and cannot be changed even though the element has not been defined with platform yet.

So in a scenario where a definition is exported by a library like:
```ts
// my-library/button
export const buttonDefinition = FASTButton.compse({
  name: `${DefaultDesignSystem.prefix}-button`
})
```

The name of the element is now set and cannot be changed. With this change a consumer could change the prefix before defining the component.
```ts
// consumer-app/main
import { buttonDefiniton } from 'my-library';

buttonDefinion.prefix = 'foo';
```

This also provides a mechanism for library authors to create utilities for consumers to easily customize the components in the library without needing to recompose every component.

Reply:
A simpler solution could be to just remove the readonly flag from `name`.

Reply:
Tagging @nicholasrice and @eisenbergeffect for their thoughts here as well

Reply:
I'm also open to some other solutions. The idea being that a library provides a similar mechanism for handling the design system as the v1 fast-components that allows consumers to alter the prefix and apply that change across a set of component definitions.

```ts
import { FooDesignSystem, buttonDefinition, checkboxDefinition } from '...';

FooDesignSystem.withPrefix('my').define(buttonDefinition, checkboxDefinition);
```
Internally, the define method would just change the prefix on each definition before calling `FASTElementDefinition.define`

Reply:
I think making `name` writable could open up all sorts of issues. Please don't do that. 

A better approach would be to add a new method to the definition, maybe call it `derive`, that takes in a set of overrides. Internally, this would create a new definition instance with the changed properties and return it without altering the original.

```ts
buttonDefinition.derive({
  name: "foo-button"
}).define();
```

Reply:
@chrisdholt @EisenbergEffect per Rob's comment, I reverted my change and added a derive method with tests.

One thing I'd like additional feedback on is if the derive method should just call `compose` rather than duplicating some of the logic.